### PR TITLE
[Easy] Add order_type to cow trades spell

### PR DESCRIPTION
**Pull request by gentrexha**
Brief comments on the purpose of your changes:

Adding an extra column to our trades spell. 

cc @bh2smith for internal review

### CODE REVIEW

File: models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql [hunk 1]
```diff
@@ -218,7 +218,8 @@ valued_trades as (
            limit_sell_amount,
            limit_buy_amount,
            valid_to,
-           flags
+           flags,
+           case when (flags % 2) = 0 then 'SELL' else 'BUY' end as order_type
```
Comment (bh2smith) [inline models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql:222]: Isn't flags a string field? I wonder how it can be evaluated mod 2.

**Review by bh2smith — APPROVED** [review]
Looks fine to me. Just confused what Modular arithmetic means for a string.

File: models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql (hunk 1 above)
Comment (gentrexha) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql:222]: That's a good point. We should make sure to have the correct column type here.

**Comment by bh2smith** [comment]
> Leaving one example of a buy and sell order here:
> 
>     * [Buy Order](https://explorer.cow.fi/orders/0x2b508299eb140a4f6a1cc28548a06bd6919468dbeb0dffde847be35943fe29ada0c605ac88359f8e50035a5708bae45f050dd3f563f603e5)
> 
>     * [Sell Order](https://explorer.cow.fi/orders/0x6a24545f2f203f353518a6b0b5b3da55f40f311a21f5e13955b5ddaabd79911ea0c605ac88359f8e50035a5708bae45f050dd3f563d91a19)

**Comment by bh2smith** [comment]
> Leaving one example of a buy and sell order here:
> 
>     * [Buy Order](https://explorer.cow.fi/orders/0x2b508299eb140a4f6a1cc28548a06bd6919468dbeb0dffde847be35943fe29ada0c605ac88359f8e50035a5708bae45f050dd3f563f603e5)
> 
>     * [Sell Order](https://explorer.cow.fi/orders/0x6a24545f2f203f353518a6b0b5b3da55f40f311a21f5e13955b5ddaabd79911ea0c605ac88359f8e50035a5708bae45f050dd3f563d91a19)

These examples don't prove to me that your query evaluates properly. Did you add these examples to the Unit Test suite?

**Comment by gentrexha** [comment]
> > Leaving one example of a buy and sell order here:
> 
> These examples don't prove to me that your query evaluates properly. Did you add these examples to the Unit Test suite?

Good point. Here's a query showing that both are correct: https://dune.com/queries/2037039?d=1

**Comment by bh2smith** [comment]
> Good point. Here's a query showing that both are correct: https://dune.com/queries/2037039?d=1

Thanks that is cool, although since flags already exists on the trades table, I was thinking something a bit [more like this](https://dune.com/queries/2038328).

We may even want to add a REAL unit test to this project for a few of those "obscure" orders:

Screenshot 2023-02-23 at 20 18 01

**Comment by gentrexha** [comment]
> Thanks that is cool, although since flags already exists on the trades table, I was thinking something a bit [more like this](https://dune.com/queries/2038328).
> 
> We may even want to add a REAL unit test to this project for a few of those "obscure" orders:

Good point. I'll add an actual unit test during the next office hours.

**Review by Hosuke — APPROVED** [review]

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**